uprotocol/uri/datamodel/UEntity.h

```diff
@@ -0,0 +1,233 @@
+     * @param name The software entity name, such as petapp or body.access.
+     * @return Returns an UEntity with the name where the version is the latest version of the service and can only be serialized
+     *      to long UUri format.
+     */
+    static UEntity longFormat(const std::string& name) {
+        return UEntity{name, std::nullopt, std::nullopt, false};
+    }
+
+    /**
+     * Static factory method for creating a uE using the software entity name, that can be used long form UUri serialisation.
+     * @param name The software entity name, such as petapp or body.access.
+     * @param version The software entity version.
+     * @return Returns an UEntity with the name and the version of the service and can only be serialized
+     *      to long UUri format.
+     */
+    static UEntity longFormat(const std::string& name,
+                              const std::optional<uint8_t> version) {
+        return UEntity{name, version, std::nullopt, false};
+    }
+
+    /**
+     * Static factory method for creating a uE using the software entity identification number, that can be used to serialize micro UUris.
+     * @param id A numeric identifier for the software entity which is a one-to-one correspondence with the software name.
+     * @return Returns an UEntity with the name where the version is the latest version of the service and can only be serialized
+     *      to long micro UUri format.
+     */
+    static UEntity microFormat(const std::optional<uint16_t> id) {
+        return UEntity{"", std::nullopt, id, false};
+    }
+
+    /**
+     * Static factory method for creating a uE using the software entity identification number, that can be used to serialize micro UUris.
+     * @param id A numeric identifier for the software entity which is a one-to-one correspondence with the software name.
+     * @param version The software entity version.
+     * @return Returns an UEntity with the name and the version of the service and can only be serialized
+     *      to micro UUri format.
+     */
+    static UEntity microFormat(const std::optional<uint16_t> id,
+                               const std::optional<uint8_t> version) {
+      return UEntity{"", version, id, false};
+    }
+
+    /**
+     * Static factory method for creating an empty software entity, to avoid working with null<br>
+     * @return Returns an empty software entity that has a blank name, no unique id and no version information.
+     */
+    static UEntity empty() {
+        static const auto EMPTY = UEntity("", std::nullopt, std::nullopt, false);
+        return EMPTY;
+    }
+
+    /**
+     * Indicates that this software entity is an empty container and has no valuable information in building uProtocol sinks or sources.
+     * @return Returns true if this software entity is an empty container and has no valuable information in building uProtocol sinks or sources.
+     */
+    [[nodiscard]] bool isEmpty() const override {
```

Comment:
what was the decision regarding nodiscard?

Reply:
It helps in finding the issue at compile time itself. I do not find a reason not to use it.